graylog2-server/src/main/java/org/graylog2/indexer/indices/Template.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.indexer.indices;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public record Template(List<String> indexPatterns, Mappings mappings, Long order, Settings settings) {
+    public static class Mappings extends HashMap<String, Object> {
+        public Mappings(Map<String, Object> m) {
+            super(m);
+        }
+    }
+
+    public static class Settings extends HashMap<String, Object> {
+        public Settings(Map<String, Object> m) {
+            super(m);
+        }
+    }
+
+    public static Template create(List<String> indexPatterns, Mappings mappings, Long order, Settings settings) {
+        return new Template(indexPatterns, mappings, order, settings);
+    }
+
+    public static Template create(String indexPattern, Mappings mappings, Long order, Settings settings) {
+        return Template.create(Collections.singletonList(indexPattern), mappings, order, settings);
+    }
```

Comment:
Do we believe that records' constructors are worse than static create methods?
If not, I'd remove both create methods, which are a wrappers that don't bring anything new.

Reply:
Static create methods can be refactored more easily, so I would generally favor them over constructors. I would not say we need to use them everywhere though, just in cases where it makes sense (with optional fields, where you do not want to pass  `null` everywhere or wrapping a type like the `String` -> `List<String>` here).